### ISSUE TITLE
Pyright crashes with RangeError

### ISSUE THREAD

**Issue by adampauls** [microsoft/pyright#6116]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
Pyright crashes when analyzing a file that imports the `trl` library. An empty file with `from trl import SFTTrainer` is enough to reproduce the stacktrace below. I think this is really a [bug in trl](https://github.com/huggingface/trl/blob/main/trl/import_utils.py#L18), but I'm not sure and I think Pyright shouldn't crash in any case ideally. 

**Code or Screenshots**
```
An internal error occurred while binding file "/Users/adpauls/sc/git/precog/anchor/.venv/lib/python3.11/site-packages/trl/import_utils.py": RangeError: The number 3.8 cannot be converted to a BigInt because it is not an integer
    at BigInt (<anonymous>)
    at _evaluateNumericBinaryOperation (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/staticExpressions.ts:222:22)
    at evaluateStaticBoolExpression (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/staticExpressions.ts:96:20)
    at Object.evaluateStaticBoolLikeExpression (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/staticExpressions.ts:186:12)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/binder.ts:1322:54)
    at C._trackCodeFlowExpressions (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/binder.ts:3360:9)
    at C.visitIf (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/binder.ts:1319:48)
    at C.visit (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:415:29)
    at C.visitNode (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at C.walk (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at C._walkStatementsAndReportUnreachable (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/binder.ts:2680:22)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/binder.ts:296:22)
    at C._createNewScope (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/binder.ts:3543:9)
    at C.bindModule (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/binder.ts:267:14)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:726:28)
    at a.timeOperation (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:44:28)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:712:38)
    at t.LogTracker.log (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/common/logTracker.ts:45:20)
    at t.SourceFile.bind (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:709:33)
    at N._bindFile (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1756:34)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1827:22)
    at a.subtractFromTime (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:56:13)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1826:41)
    at a.importLookup [as timeOperation] (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at resolveAliasDeclaration (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/declarationUtils.ts:268:28)
    at resolveAliasDeclaration (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20700:16)
    at forEach (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20865:17)
    at Array.forEach (<anonymous>)
    at getEffectiveTypeOfSymbolForUsage (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20831:15)
    at getTypeOfName (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:4391:43)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:997:30)
    at a.getTypeOfExpression [as timeOperation] (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at isCallNoReturn (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1503:46)
    at isFlowNodeReachableRecursive (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1174:44)
    at isFlowNodeReachableRecursive (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1198:29)
    at isFlowNodeReachable (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1250:20)
    at getTypeFromFlowNode (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:573:33)
    at Object.getTypeFromCodeFlow (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1094:20)
    at getFlowTypeOfReference (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19123:25)
    at getTypeOfName (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:4437:48)
    at getTypeOfExpression (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:997:30)
    at forEach (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:15822:29)
    at Array.forEach (<anonymous>)
    at getTypeOfClass (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:15805:28)
    at getTypeForDeclaration (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20190:39)
    at getInferredTypeOfDeclaration (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20492:30)
    at getAliasedSymbolTypeForName (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:18535:16)
    at evaluateTypesForImportFromAs (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:18231:26)
    at callback (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19021:21)
    at a.evaluateTypesForStatement [as timeOperation] (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at forEach (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:1476:33)
    at Array.forEach (<anonymous>)
    at U.visitImportFrom (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:1475:26)
    at U.visit (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:424:29)
    at U.visitNode (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at U.walk (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at U.walk (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:266:19)
    at forEach (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:924:22)
    at Array.forEach (<anonymous>)
    at U.walkMultiple (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:922:15)
    at U.walk (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:917:18)
    at U.walk (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:266:19)
    at U._walkStatementsAndReportUnreachable (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2651:18)
    at U.check (/Users/adpauls/.cache/pyright-python/1.1.330/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:245:14)
```

**VS Code extension or command-line**
Command line, version 1.330

**Comment by adampauls** [comment]
I can confirm that locally editing `import_utils.py` to properly check the sys version fixes the issue.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

And yes, this is a bug in the library as well. You may want to report it to the maintainers.

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
